### PR TITLE
Add date prop to DateControl

### PR DESCRIPTION
**Pull request by hursey013**
Fixes https://github.com/18F/e-QIP-prototype/issues/1167 and prepares for https://github.com/18F/e-QIP-prototype/issues/985

The change to `DateControl` is relatively straightforward, most of the effort here is refactoring the tests to account for the change.

### CODE REVIEW

File: src/components/Section/Relationships/RelationshipStatus/CivilUnion.test.jsx [hunk 1]
```diff
@@ -4,8 +4,9 @@ import configureMockStore from 'redux-mock-store'
 import { Provider } from 'react-redux'
 import CivilUnion from './CivilUnion'
 
-const mountComponent = (mockStore, Component, props) => {
-  const store = mockStore({ application: { AddressBooks: {} }})
+const mountComponent = (Component, props = {}) => {
```
Comment (gidjin) [inline src/components/Section/Relationships/RelationshipStatus/CivilUnion.test.jsx]: Why is this called mountComponent and the others are call createComponent? I'd recommend a consistent name if possible for this function in all the tests.

File: src/components/Section/Financial/Gambling/Gambling.test.jsx [hunk 2]
```diff
@@ -89,17 +105,53 @@ describe('The gambling component', () => {
...
-    const expected = {
-      HasGamblingDebt: { value: 'No' },
-      List: { branch: {}, items: [{}] }
-    }
-    const component = mount(<Gambling {...expected} />)
-    expect(component.instance().fancyNumber(1000)).toEqual('1,000')
-    expect(component.instance().fancyNumber(10000)).toEqual('10,000')
+    const tests = [
+      {
+        HasGamblingDebt: { value: 'Yes' },
```
[the remark this answers is not in the document]
Comment (ryanhofdotgov) [reply src/components/Section/Financial/Gambling/Gambling.test.jsx]: If you look at the original test, it was testing the fancyNumber function in a weird way – checking a gambling debt value when you had no gambling debt. Now it tests the component with sensical test input data and more direct output value.

File: src/components/Section/Foreign/Activities/IndirectActivity/IndirectActivity.test.jsx [hunk 3]
```diff
@@ -1,19 +1,36 @@
...
-    const component = mount(<IndirectActivity />)
+    const component = createComponent()
     expect(component.find('.indirect').length).toBe(1)
   })
 
   it('Updates with yes', () => {
     let updates = 0
-    const onUpdate = () => {
-      updates++
+    const expected = {
```
[the remark this answers is not in the document]
Comment (ryanhofdotgov) [reply src/components/Section/Foreign/Activities/IndirectActivity/IndirectActivity.test.jsx]: Just glancing through the diff quickly, I would tend to disagree. Most of the tests have several properties/methods in `expected` and this creates a homogeneity in the structure across them.

File: src/components/Section/Relationships/RelationshipStatus/CivilUnion.test.jsx (hunk 1 above)
Comment (ryanhofdotgov) [reply src/components/Section/Relationships/RelationshipStatus/CivilUnion.test.jsx]: Well, while they perform similar functions, these 2 different functions have different function signatures – `mountComponent` takes the component as an argument, `createComponent` does not.

While the 3 components that use `mountComponent` could be refactored further, the current PR seems to make a good tradeoff between changing the bare minimum required to the existing tests to make the `DateControl` change pass and more sweeping changes across the entire front-end test suite.

**Pull request merged**